### PR TITLE
fix: Multi-word custom endpoint not respecting route format

### PR DESCRIPTION
**Pull request by sebasjimenez10**
This PR fixes an issue I experienced when working with multi-word custom endpoints. If you have a multi-word custom endpoint and you also have the `self.route_format = :dasherized_route` config in your base class, upon calling that `custom_endpoint` JsonApiClient doesn't respect the configuration selected and will use the underscored version of the endpoint name.

For instance:

Given,

```ruby
class Pet < TestResource
  self.site = "http://example.com/"
  self.route_format = :dasherized_route

  custom_endpoint :related_pets, on: :member, request_method: :get
  custom_endpoint :vip_pets, on: :collection, request_method: :get
end
```

When calling `Pet.vip_pets` or `pet = Pet.new(...); pet.related_pets` the resulting endpoint being called will be either:
`http://example.com/pets/vip_pets` or `http://example.com/pets/:id/related_pets`

instead of `http://example.com/pets/vip-pets` or `http://example.com/pets/:id/related-pets`

Same issue applies for `camelized_routes`.

This PR fixes this by checking the `route_format` configuration attribute and making sure it is converted to the appropriate endpoint name, using `ActiveSupport` helper methods.

The methods defined in the instance of the class will still be underscored, so that we can still do `Pet.vip_vets`.

Happy to address any concerns or comments, thanks!

### CODE REVIEW

**Comment by sebasjimenez10** [comment]
Hey @gaorlov 👋 giving this one a bump for you to take a look 🙏

**Comment by esbanarango** [comment]
What's the current blocker for merging this one? 🙏

**Comment by gaorlov** [comment]
Hello! Thanks for taking the time to contribute! I'll take a closer look tomorrow. I'm the meantime, can you please add an entry to the changelog?

Thanks!

Greg

**Comment by sebasjimenez10** [comment]
@gaorlov updated the changelog! 🙏

**Comment by sebasjimenez10** [comment]
@gaorlov small bump! 🙂

**Comment by gaorlov** [comment]
@sebasjimenez10 sorry about the delay. For reasons i don't yet understand the automated testing has stopped working. I'm working with the owners of the `JsonAPiClient` org to resolve this issue and will let you know when we can start merging again.

I appreciate your patience!

Greg

**Comment by sebasjimenez10** [comment]
Hey @gaorlov, just dropping by and giving this one a bump! 🙏

**Comment by sebasjimenez10** [comment]
Hey @gaorlov, hope you're doing good! Just wanted to bump this one!

**Comment by esbanarango** [comment]
🙏

File: lib/json_api_client/resource.rb [hunk 1]
```diff
@@ -348,6 +350,17 @@ def _build_connection(rebuild = false)
           yield(conn) if block_given?
         end
       end
+
+      def _name_for_route_format(name)
```
Comment (gaorlov) [inline lib/json_api_client/resource.rb:354]: Can this be a method on `name`? like `name.to_route(route_format)`?

**Review by gaorlov — COMMENTED** [review]
Looks great! thanks for contributing!
Please merge in main to get tests running in the PR
Thanks!

File: lib/json_api_client/resource.rb (hunk 1 above)
Comment (sebasjimenez10) [reply lib/json_api_client/resource.rb:354]: @gaorlov name here is a Symbol, that would require a big refactor in other places as well and I believe that to be out of scope for this PR. But happy to discuss further to continue contributing 🙏

**Comment by sebasjimenez10** [comment]
@gaorlov done! Just rebased on top of `master` (main). Tests are looking good. Please let me know if there is anything else I need to do!

**Comment by sebasjimenez10** [comment]
@gaorlov small bump!

**Comment by sebasjimenez10** [comment]
@gaorlov tiny bump on this one! 🙏

File: lib/json_api_client/resource.rb (hunk 1 above)
Comment (gaorlov) [reply lib/json_api_client/resource.rb:354]: Could you do like
```ruby
module Routable
  def to_route(format: :dasherized_route)
    case format
    when :dasherized_route
      to_s.dasherize
    when :camelized_route
      to_s.camelize(:lower)
    else
      self
    end
  end
end
 
class Symbol
  include Routable
end
```
and then you can have `name.to_route(route_format)` and not have to do anything else to the code

**Comment by gaorlov** [comment]
thank you for you contributions and patience!
[1.23.0](https://rubygems.org/gems/json_api_client/versions/1.23.0) is now live with your changes

**Pull request merged**